### ISSUE TITLE
Selected Properties discard subsequent changes [v4.7 & v4.8-beta]

### ISSUE THREAD

**Issue by Matthew-QC** [microsoft-search/pnp-modern-search#2401]
**Version used**
4.7 & 4.8.0-beta.1

**Describe the bug**
I apologise if this has been noted in another issue, I tried to search through discussions and issues but didn't find anything. When adding a Search Result web part to a site page, the "Selected Properties" option allows configuration up until the first page publish. Changing the "Selected Properties" in subsequent publishes are ignored - reverts back to the properties selected on the first publish.

**To Reproduce**
1. Add "Search Results" & "Search Box" Web Parts to Site Page
2. Configure "Search Results" as follows (all other settings are defaults):
    - Data Source: Sharepoint
    - Selected Properties [Set 1]: Default + Color + ModifiedBy <example only, choose any configuration of properties you like>
    - Layout: Debug (for ease of testing)
    - Connections: Use input query text > PnP - Search Box > Search query
4. Publish Page
5. Perform Search in PnP Search Box as usual
6. Note that the Properties selected in [Set 1] are displayed as expected
7. Edit Page > Edit Search Results
    - Selected Properties [Set 2]: [Set 1] - Color + ModifiedById <again this is purely as an example - Note "Color" is removed>
    - Leave all other settings as before
8. Re-Publish Page
9. Perform the same search in the Search Box
10. Note that Properties displayed are still from [Set 1]
11. Edit Page > Edit Search Results
    - Selected Properties showing [Set 1], [Set 2] has been discarded

**Expected behavior**
Selected Properties save changes in subsequent publishes of the page and don't revert to the originally published set of properties.
E.g. [Set 2] properties are displayed in the second search, not [Set 1]. 

**Screenshots**
Happy to provide if needed. 

**Desktop (please complete the following information):**
 - Browser: Edge Chromium

**Additional context**
I have replicated this issue in both 4.7 and 4.8-beta.1 using a full reinstall of the app. I have also tested and observed the same behaviour for existing and completely fresh Sharepoint sites - with only a Home and Test page. Note that Teams Sites are used, I haven't tested a Communication Site.

Edit (Additional Information): 
Deleting and adding a new "Search Results" Webpart will save the new Set on creation, but then exhibit the same behaviour for any future edits. Meaning that for each time you want to change the "Selected Properties" option, you have to delete and entirely recreate the "Search Results" webpart.

**Comment by Matthew-QC** [comment]
Found Issue [#968](https://github.com/microsoft-search/pnp-modern-search/issues/968) which seems to refer to this same error, however, it was marked as an incomplete submission.

**Comment by Matthew-QC** [comment]
Hi @FranckyC, thanks for looking into this and trying to replicate the issue. For my own sanity I have just now uninstalled the app (deleting from all first- and second-stage recycle bins) and reuploaded 4.8-beta.1 (note that when I upload the app, I select "activate only" since I want to control which sites have the app added to it - I wouldn't think this would make a difference?). 

I then created a completed fresh Communication site this time (in case it made a difference) and tested on a new Test page on this site. Unfortunately, I still observe the mentioned issue using the same steps listed above (also used different properties this time, and I can confirm it doesn't seem to matter which properties are selected). 

If there are any debug logs you would like me to collect, I'm happy to assist however I can. Also, just in case I'm a total dud, there isn't some "save" button on the Search Results Web Part that I'm missing is there? I see them for other options, but not "Selected Properties" and closing/reopening the Web Part options pane seems to hold the newly selected properties, it is only after re-publishing the page that the settings are lost.

**Comment by wobba** [comment]
I built the release/4.8.0 branch and tested and cannot repro. When I remove props first added, they are no longer persisted. @Matthew-QC can you go to web part props page 4 and click `Edit Properties` and see what the list of props look like?

**Comment by Matthew-QC** [comment]
Hi @wobba, thank you as well for looking into this. I've done what you said and have noticed that after the initial publishing, the "Edit Properties" json doesn't reflect further changes made to the "Selected Properties" field, which would be what is causing the issue on my end. ~~I've linked a screen capture showing how I'm setting it up, and what I'm observing.~~

While writing this I had an idea. It appears that my adblocker (ublock origin) is interfering with that particular field updating in the "Edit Properties" json. Disabling the adblocker for the site allows me to make changes as I would normally expect. I hadn't noticed it interfering with any other Web Part properties in the past, so I initially didn't consider it. 

It might take a little more digging to find out why it is being blocked, but it's a simple fix to disable it for SharePoint.

**Comment by wobba** [comment]
@Matthew-QC can we close this one then?

**Comment by Matthew-QC** [comment]
This one can be closed for now, although perhaps it is worth noting that some browser extensions may interfere with the how the web part stores configuration properties. But whitelisting the SharePoint site is an easy enough work-around for the time being. 

Thank you for taking the time to investigate this for me.
